### DISCUSSION TITLE
Method of exception throwing in ESBMC

### DISCUSSION THREAD

**Discussion by fbrausse** [esbmc/esbmc#702] [General]
In cases where array sizes or irep2 type errors are involved, we throw dynamically exceptions and catch by typed pointer. This seems strange to me, as it is
1. non-idiomatic C++, where exception objects are thrown directly, and
2. wasteful dynamic memory allocation.

In particular, this concerns the following exception types:
```c
array_type2t::dyn_sized_array_excparrsize
array_type2t::inf_sized_array_excp
type2t::symbolic_type_excp
```
Why has this approach been adopted? Should we replace those by throwing plain objects?

**Comment by mikhailramalho** [comment]
It was a design decision lost to time. I'm fine with removing it.

**Reply by fbrausse** [reply]
Many thanks, Mikhail! I'll prepare a patch.
